packages/block-library/src/freeform/edit.js

```diff
@@ -227,6 +230,7 @@ function ClassicEdit( {
 				onReadyStateChange
 			);
 			wp.oldEditor.remove( `editor-${ clientId }` );
+			didMount.current = false;
```

Comment:
Good find!

Reply:
We don't do this in a lot of places. Is this necessary in practice?

Reply:
It depends. A cleanup could be necessary when you have two connected effects like this and want to ensure that one only does its job after the initial mount.

Here's an example: https://codesandbox.io/p/sandbox/heuristic-rhodes-stw52g?file=%2Fsrc%2FApp.js